### DISCUSSION TITLE
Is there a batch morphometric extraction script (xls file)?

### DISCUSSION THREAD

**Discussion by Fragglepuss** [neuropoly/axondeepseg#505] [Help needed] [answered]
I need to export morphometrics from all of my images to xls files. Right now my process is to open fsleyes, load my image and mask, then click the morphometric extraction button. Unfortunately I have 150 images so this is a pretty time consuming process. I've used the morphometrics jupyter notebook but that doesn't give a detailed breakdown like the xls file on fsleyes. 

Is there a script that will allow me to run morphometric extraction and export to an xls file on multiple images? This would save me a lot of time. If not, is there a script that would allow me to export to an xls file from my terminal rather than having to open fsleyes to get one? 

Thanks!

**Comment by mathieuboudreau** [comment, marked as answer]
Hi @Fragglepuss,

Thanks for reaching out! Yes, that's certainly a feature that would be useful for users like you.

We are in fact currently implementing that feature already, it's in PR #434. We'll move this up the priority list and will let you know when it has been merged.

**Reply by Fragglepuss** [reply]
Great. Thanks a lot and great job.

**Reply by vasudev-sharma** [reply]
Hey @Fragglepuss, one question for you: do you have all the 150 images in the same folder/directory or different directories?

**Reply by Fragglepuss** [reply]
Hi @vs74. I have a copy of my directory with each image in a different folder. Between manual segmentation, training, desnoising, processing the images for analysis outside of ADS, etc., I have a bunch of copies in just about every organizational structure they could be in. I'll test the batch segmentation today and report back.

**Reply by vasudev-sharma** [reply]
@Fragglepuss, we've merged the CLI feature for generating morphometrics in ADS. Kindly refer to the [documentation](https://axondeepseg.readthedocs.io/en/latest/documentation.html#morphometrics) to look out for how to get started with it. For it to get working, I suggest doing a fresh clone and install of ADS in a virtual environment by referring the [installation](https://axondeepseg.readthedocs.io/en/latest/documentation.html#installation) doc.

If you do encounter any issues, let us know.
